platforms/core-configuration/model-core/src/main/java/org/gradle/internal/instantiation/generator/AsmBackedClassGenerator.java

```diff
@@ -984,6 +984,14 @@ public void addLazyGroovySupportSetterOverloads(PropertyMetadata property, Metho
                 return;
             }
 
+            // TODO: fix/remove in 9.0
```

Comment:
Hopefully, this is a temporary workaround until we have a conclusion on `url = "string"` vs `url = uri(...)`

Reply:
🤔 Note that this changes the behaviour for all properties. Can we maybe limit this? Maybe to just add this workaroudn for `Property<Url>` or even to handle that in the `Property` type